### PR TITLE
fix: Override props with injections

### PR DESCRIPTION
**Pull request by ajdaniel**
If you have a prop and an injection with the same name (perhaps you defined it in a static propTypes var) then the injection will take precedence and override any props with the same name.
I also added @types/lodash to the build because I was having problems with the typings for lodash until I did that (I am happy to modify this)

### CODE REVIEW

**Comment by ajdaniel** [comment]
If you prefer more tests please let me know. I could also add some documentation to the readme if needed.

**Comment by ajdaniel** [comment]
@bcherny Hey, could you take a look into this please?

File: package.json [hunk 1]
```diff
@@ -71,7 +72,7 @@
     "watchify": "^3.10.0"
   },
   "dependencies": {
-    "@types/angular": ">=1.5",
+    "@types/angular": "^1.6.43",
```
Comment (bcherny) [inline package.json]: Please don't change this

File: package.json [hunk 2]
```diff
@@ -46,6 +46,7 @@
     "@types/angular-mocks": "^1.5.11",
     "@types/jasmine": "^2.8.6",
     "@types/jquery": "^3.3.0",
+    "@types/lodash": "^4.14.104",
```
Comment (bcherny) [inline package.json]: Same here, is this used?

Comment (bcherny) [reply package.json]: What exactly were the problems you were having with the build?

**Comment by edmorley** [comment]
Hi @ajdaniel - I don't suppose you have a chance to address the review comments any time soon? :-)

**Comment by ajdaniel** [comment]
My apologies, I missed the notification for this, I will take a look very soon

**Comment by ajdaniel** [comment]
Okay so I reverted my changes to `package.json` and the tests run okay, so I've updated my PR commit now

**Comment by ajdaniel** [comment]
Do you think this feature should be highlighted in the documentation?

**Comment by ajdaniel** [comment]
I just pushed an amend to my commit that has a brief mention in the readme.

**Comment by bcherny** [comment]
Thanks for the contribution @ajdaniel!

**Comment by bcherny** [comment]
Published @4.0.0.

**Pull request merged**